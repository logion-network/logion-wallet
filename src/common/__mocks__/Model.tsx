import {
    fetchProtectionRequests,
    preVoid,
    isGrantedAccess,
} from './ModelMock';

export {
    fetchProtectionRequests,
    preVoid,
    isGrantedAccess,
};
