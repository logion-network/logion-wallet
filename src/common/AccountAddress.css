.AccountAddress {
    width: 100%;
    display: flex;
    flex-direction: row;
}

.AccountAddress .icon {
    height: 75px;
    width: 75px;
    border-radius: 36px;
    margin-left: 20px;
    margin-right: 20px;
    font-size: 45px;
    font-weight: bold;
    color: white;
    padding-left: 18px;
    flex-shrink: 0;
}

.AccountAddress .text {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 75px;
    width: calc(100% - 180px);
}

.AccountAddress.disabled {
    cursor: default;
}

.AccountAddress.disabled .icon,
.AccountAddress.disabled .text {
    opacity: 0.5;
}

.AccountAddress .text .balance {
    max-width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
}

.AccountAddress .text .balance .symbol {
    font-size: 0.7rem;
}

.AccountAddress .text .name {
    font-size: 1.1rem;
    font-weight: bold;
    max-width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
}

.AccountAddress .text .address {
    font-size: 0.9rem;
    max-width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
}

.AccountAddress .login {
    padding-left: 20px;
    padding-right: 20px;
    flex-shrink: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.AccountAddress .login .login-button {
    height: 20px;
    width: 20px;
    border-radius: 10px;
    line-height: 1;
    cursor: pointer;
    position: relative;
    z-index: 1;
}

.AccountAddress .login .login-button img {
    margin-left: 3px;
}
