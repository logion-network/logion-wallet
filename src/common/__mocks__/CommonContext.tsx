import { 
    useCommonContext as useCommonContextMock,
} from './CommonContextMock';

export const useCommonContext = useCommonContextMock;
export { ExpectNewTransactionStatus } from "./CommonContextMock";
