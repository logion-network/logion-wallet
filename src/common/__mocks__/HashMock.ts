export let sha256Hex: jest.Mock<any, any> = jest.fn();

export function setExpectedHash(expectedHash: string) {
    sha256Hex = jest.fn().mockResolvedValue(expectedHash)
}
