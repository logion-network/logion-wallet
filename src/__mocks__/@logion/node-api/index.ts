import { ApiPromise } from "src/__mocks__/PolkadotApiMock";
export * from "@logion/node-api/dist/UUID";

export function buildApi() {
    return new ApiPromise();
}

export * from "./dist/LogionLoc";
