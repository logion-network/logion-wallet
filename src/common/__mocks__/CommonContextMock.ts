import { COLOR_THEME, threeLegalOfficers } from '../TestData';
import { CommonContext, Viewer } from "../CommonContext";
import { BalanceState } from "@logion/client/dist/Balance.js";

export let setColorTheme = jest.fn();

export let refresh = jest.fn();

export let balanceState: BalanceState | undefined = {
    balances: [],
} as unknown as BalanceState;

export function setBalanceState(value: BalanceState | undefined) {
    balanceState = value;
}

let viewer: Viewer = "User";

export function setViewer(_viewer: Viewer) {
    viewer = _viewer;
}

export enum ExpectNewTransactionStatus {
    IDLE,
    WAITING_NEW_TRANSACTION,
    DONE
}

export function useCommonContext() {
    const commonContext:Partial<CommonContext> = {
        balanceState,
        accountsBalances: {},
        colorTheme: COLOR_THEME,
        setColorTheme,
        refresh,
        nodesUp: [],
        nodesDown: [],
        availableLegalOfficers: threeLegalOfficers,
        viewer,
        backendConfig: () => ({ features:{ iDenfy: false, vote: false }}),
        expectNewTransactionState: { status: ExpectNewTransactionStatus.IDLE, refreshCount: 0 },
    };
    return commonContext;
}

