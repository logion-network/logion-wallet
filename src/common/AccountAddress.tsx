import { CoinBalance } from "@logion/node-api";
import { CSSProperties } from 'react';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';

import { Account } from './types/Accounts';
import { useCommonContext } from './CommonContext';
import Icon from './Icon';

import './AccountAddress.css';
import { Spinner } from "react-bootstrap";

export interface Props {
    balance?: CoinBalance | null;
    account: Account;
    disabled: boolean;
    login?: () => void;
}

export default function AccountAddress(props: Props) {
    const { colorTheme } = useCommonContext();

    let style: CSSProperties = {};
    const backgroundIcon = colorTheme.accounts.legalOfficerIcon;
    let foregroundIcon;
    if(props.account.isLegalOfficer) {
        style['backgroundImage'] = `url("${process.env.PUBLIC_URL}/assets/${backgroundIcon.id}.svg")`;
        style['backgroundRepeat'] = 'no-repeat';
        style['backgroundPositionX'] = '7px';
        style['backgroundPositionY'] = 'center';

        foregroundIcon = "polkadot-account-white";
    } else {
        style['backgroundColor'] = colorTheme.accounts.iconBackground;
        foregroundIcon = props.account.accountId.type === "Ethereum" ? "metamask" : "polkadot-account";
    }

    return (
        <div className={ "AccountAddress" + ( props.disabled ? " disabled" : "" ) }>
            <div
                className="icon"
                style={ style }
            >
                <Icon icon={{ id: foregroundIcon }} width="40px" />
            </div>
            <div
                className="text"
                style={{
                    justifyContent: props.balance === undefined ? 'space-around' : 'space-between'
                }}
            >
                <div
                    className="name"
                    style={{
                        color: colorTheme.accounts.foreground,
                    }}
                >
                    { props.account.name }
                </div>
                <div
                    className="address"
                    style={{
                        color: colorTheme.accounts.foreground,
                    }}
                >
                    <OverlayTrigger
                      placement="bottom"
                      delay={ 500 }
                      overlay={
                        <Tooltip id={`tooltip-${props.account.accountId.toKey()}`}>
                          { props.account.accountId.address }
                        </Tooltip>
                      }
                    >
                      <span>{ props.account.accountId.address }</span>
                    </OverlayTrigger>
                </div>
                {
                    props.balance !== undefined &&
                    <div
                        className="balance"
                        style={{
                            color: colorTheme.accounts.foreground,
                        }}
                    >
                        <span className="amount">{ props.balance === null ? <Spinner size="sm"/> : props.balance.balance.coefficient.toFixedPrecision(2) }</span>{" "}
                        <span className="symbol">{ (props.balance?.balance.prefix.symbol || "") + (props.balance?.coin.symbol || "LGNT") }</span>
                    </div>
                }
            </div>
            {
                props.login !== undefined && props.account.token === undefined &&
                <div
                    className="login"
                >
                    <div
                        className="login-button"
                        onClick={ props.login }
                        style={{
                            border: `1px solid ${colorTheme.accounts.foreground}`
                        }}
                        role="button"
                        aria-label="login button"
                    >
                        <Icon
                            icon={{ id: "forward", hasVariants: true }}
                            colorThemeType={ colorTheme.type }
                        />
                    </div>
                </div>
            }
        </div>
    );
}
