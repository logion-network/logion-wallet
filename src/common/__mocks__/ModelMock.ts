import { LocRequest } from "@logion/client";
import { CreateLocRequest } from "../Model";

export let fetchProtectionRequests = jest.fn();

export function setFetchProtectionRequests(mockFn: any) {
    fetchProtectionRequests = mockFn;
}

export async function createLocRequest(axios: any, request: CreateLocRequest): Promise<LocRequest> {
    return Promise.resolve({
        ...request,
        createdOn: "2022-01-20T09:24:00.000",
        description: "Test",
        files: [],
        id: "671e65cf-42c3-435b-a222-be008e7fb3d1",
        links: [],
        metadata: [],
        status: "OPEN",
        verifiedThirdParty: false,
        selectedIssuers: [],
    });
}

export const preVoid = jest.fn();

export function resetDefaultMocks() {
    preVoid.mockResolvedValue(undefined);
}

export function isGrantedAccess() {
    return true;
}
