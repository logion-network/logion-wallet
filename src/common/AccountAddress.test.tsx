import { Numbers } from "@logion/node-api";
import { mockValidPolkadotAccountId } from 'src/__mocks__/LogionMock';
import { shallowRender } from '../tests';

import AccountAddress from './AccountAddress';

describe("AccountAddress", () => {

    it("renders", () => {
        const result = shallowRender(
            <AccountAddress
                balance={{
                    available: new Numbers.PrefixedNumber("2", Numbers.NONE),
                    total: new Numbers.PrefixedNumber("2", Numbers.NONE),
                    reserved: new Numbers.PrefixedNumber("0", Numbers.NONE),
                    coin: {
                        id: "lgnt",
                        symbol: "LGNT"
                    },
                    level: 1
                }}
                account={{
                    name: "Name 1",
                    accountId: mockValidPolkadotAccountId("address1"),
                    isLegalOfficer: false,
                }}
                disabled={ false }
            />
        );
        expect(result).toMatchSnapshot();
    });
    
});
