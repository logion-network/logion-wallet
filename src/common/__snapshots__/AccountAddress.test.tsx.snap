// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="AccountAddress"
>
  <div
    className="icon"
    style={
      Object {
        "backgroundColor": "#3b6cf4",
      }
    }
  >
    <Icon
      icon={
        Object {
          "id": "polkadot-account",
        }
      }
      width="40px"
    />
  </div>
  <div
    className="text"
    style={
      Object {
        "justifyContent": "space-between",
      }
    }
  >
    <div
      className="name"
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      Name 1
    </div>
    <div
      className="address"
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      <OverlayTrigger
        defaultShow={false}
        delay={500}
        overlay={
          <Tooltip
            id="tooltip-Polkadot:address1"
            placement="right"
          >
            address1
          </Tooltip>
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <span>
          address1
        </span>
      </OverlayTrigger>
    </div>
    <div
      className="balance"
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      <React.Fragment>
        <InlineAmount
          amount={
            PrefixedNumber {
              "_prefix": Object {
                "symbol": "",
                "tenExponent": 0,
              },
              "_scientificNumber": ScientificNumber {
                "_normalized": NormalizedNumber {
                  "_decimalPart": "",
                  "_integerPart": "2",
                  "_negative": false,
                  "_normalized": "2.",
                },
                "_tenExponent": 0,
              },
            }
          }
          coin={
            Object {
              "id": "lgnt",
              "symbol": "LGNT",
            }
          }
        />
         
      </React.Fragment>
    </div>
  </div>
</div>
`;
